### PR TITLE
fix(rnd): Fix broken save feature on Agent Builder

### PR DESCRIPTION
**Pull request by majdyz**
### Background

Save feature is currently broken
image

### Changes 🏗️

* Fixed agent save feature.
* Added agent version information on the save input box.
* Improved the save experience by routing agent flowID after saving.
* Fixed graph deep equal comparison, currently it always saves the same graph.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

**Review by ntindle — APPROVED** [review]

File: rnd/autogpt_builder/src/hooks/useAgentGraph.ts [hunk 1]
```diff
@@ -608,7 +614,7 @@ export default function useAgentGraph(
           }));
 
         return {
-          id: node.id,
+          id: node.data.backend_id,
```
Comment (Ujjawal-Anand) [inline rnd/autogpt_builder/src/hooks/useAgentGraph.ts:617]: This is breaking the execution of blocks. I believe this should be switched to existing value

Comment (majdyz) [reply rnd/autogpt_builder/src/hooks/useAgentGraph.ts:617]: Thank you for flagging this!
I have created a follow-up PR to fix it, let me know if the issue persists:
https://github.com/Significant-Gravitas/AutoGPT/pull/8054

File: rnd/autogpt_builder/src/hooks/useAgentGraph.ts [hunk 2]
```diff
@@ -637,11 +643,8 @@ export default function useAgentGraph(
         nodes: formattedNodes,
         links: links,
       };
-
-      if (savedAgent && deepEquals(payload, savedAgent)) {
-        console.debug(
-          "No need to save: Graph is the same as version on server",
-        );
+      if (savedAgent && deepEquals(payload, savedAgent, true)) {
+        console.warn("No need to save: Graph is the same as version on server");
```
Comment (Pwuts) [inline rnd/autogpt_builder/src/hooks/useAgentGraph.ts:647]: This isn't a `warn` level message

File: rnd/autogpt_builder/src/hooks/useAgentGraph.ts [hunk 3]
```diff
@@ -637,11 +643,8 @@ export default function useAgentGraph(
         nodes: formattedNodes,
         links: links,
       };
-
-      if (savedAgent && deepEquals(payload, savedAgent)) {
-        console.debug(
-          "No need to save: Graph is the same as version on server",
-        );
+      if (savedAgent && deepEquals(payload, savedAgent, true)) {
```
Comment (Pwuts) [inline rnd/autogpt_builder/src/hooks/useAgentGraph.ts:646]: What keys are being added on save that we need the `allowMissingKeys`?

File: rnd/autogpt_builder/src/hooks/useAgentGraph.ts (hunk 2 above)
Comment (majdyz) [reply rnd/autogpt_builder/src/hooks/useAgentGraph.ts:647]: Unless we fix the UX for saving, not having this warning is super confusing.

File: rnd/autogpt_builder/src/hooks/useAgentGraph.ts (hunk 3 above)
Comment (majdyz) [reply rnd/autogpt_builder/src/hooks/useAgentGraph.ts:646]: Any random field being added from the server can break this.
Currently:
id, subgraph, version,

Comment (majdyz) [reply rnd/autogpt_builder/src/hooks/useAgentGraph.ts:646]: Also additional fields on nodes too I don't really remember what those are

Comment (majdyz) [reply rnd/autogpt_builder/src/hooks/useAgentGraph.ts:646]: @Pwuts I think it's safer to just explicitly define what needs to be compared, I have reverted this change here:
https://github.com/Significant-Gravitas/AutoGPT/pull/8054/files#diff-7e0162cb7a3b78d205d099d20b98cdf25239e6faa2113d6843520a8e21797296R33-R34

**Pull request merged**